internal/providers/terraform/azure/storage_table.go

```diff
@@ -25,19 +24,16 @@ func newStorageTable(d *schema.ResourceData) schema.CoreResource {
 	region := d.Region
 
 	accountReplicationType := "LRS"
-	accountKind := "StorageV2"
+	hasCustomerManagedKey := false
 
 	if len(d.References("storage_account_name")) > 0 {
 		storageAccount := d.References("storage_account_name")[0]
+		accountReplicationType = storageAccount.Get("account_replication_type").String()
+		hasCustomerManagedKey = storageAccount.Get("customer_managed_key").Exists()
```

Comment:
We need to support both the inline `customer_managed_key` as defined [here](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account.html#customer_managed_key-1) as well as the one that's in a [separate resource](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account_customer_managed_key)